app/src/main/java/com/example/androidcourselevel5/presentation/ui/AuthorizationActivity.kt

```diff
@@ -1,12 +1,265 @@
+        }
+
+        with(binding) {
+
+            etEmailFiled.doOnTextChanged { text, _, _, _ ->
+                text?.let {
+                    if (text.isEmpty()) {
+                        tvEmailFiledHelper.clear()
+                    } else if (!authorizationViewModel.validateEmail(text.toString())) {
+                        showEmailErrorMessage()
+                    } else {
+                        tvEmailFiledHelper.clear()
+                    }
+                }
+            }
+
+            etPasswordField.doOnTextChanged { text, _, _, _ ->
+                text?.let {
+                    if (text.isEmpty()) {
+                        tvPasswordFiledHelper.clear()
+                    } else {
+                        val validationResult =
+                            authorizationViewModel.validatePassword(text.toString())
+                        if (validationResult != getString(R.string.validate_success)) {
+                            showPasswordErrorMessage(validationResult)
+                        } else {
+                            tvPasswordFiledHelper.clear()
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun launchTransitionToProfile(userData: UserData) {
+        authorizationViewModel.saveAllUserDataToDataStorage(
+            userData = userData, password = binding.etPasswordField.text.toString(),
+            checkboxIsChecked = binding.checkBoxAuthorizationRememberMe.isChecked)
+
+        startActivity(Intent(this, MainActivity::class.java))
+        showTransitionAnimation()
+        binding.etEmailFiled.clear()
+        binding.etPasswordField.clear()
+    }
+
+
+    private fun setListeners() {
+
+        with(binding) {
+
+            btnAuthorizationRegister.setOnClickListener {
+                if (authorizationViewModel.validateEmailAndPassword(
+                        email = etEmailFiled.text.toString(),
+                        password = etPasswordField.text.toString())) {
+
+                    launchAuthorisation()
+                    authorizationViewModel.pressRegistrationButton()
+
+                } else {
+                    Toast.makeText(this@AuthorizationActivity,
+                        getString(R.string.empty_password_or_email_fields), Toast.LENGTH_SHORT).show()
+                }
+            }
+
+            tvAuthorizationSignInText.setOnClickListener {
+                if (authorizationViewModel.validateEmailAndPassword(
+                        email = etEmailFiled.text.toString(),
+                        password = etPasswordField.text.toString())) {
+                    launchAuthorisation()
+                    authorizationViewModel.pressSignInButton()
+
+                } else {
+                    Toast.makeText(this@AuthorizationActivity,
+                        getString(R.string.empty_password_or_email_fields), Toast.LENGTH_SHORT).show()
+                }
+            }
```

Comment:
This action must navigating to Authorization screen (SignIn). SignIn must be separated screen. This screen is not Authorization but Registration because has such layout where contains button "Register".